### DISCUSSION TITLE
Missing parts of dashed line, line and road

### DISCUSSION THREAD

**Discussion by PhungVanHoa** [mapsforge/vtm#952] [General]
Hi, we implement mapsforge/vtm library in our project and when using online vector maps from **maptiler** (OpenMapTilesMvtTileSource: https://api.maptiler.com/tiles/v3), we got rendering Issues, missing parts of dashed line and overlapping line (or road) above water
![image](https://user-images.githubusercontent.com/16158238/184895046-67bcf9be-e96f-42fe-897e-700271739f9d.png)
![image](https://user-images.githubusercontent.com/16158238/184895148-9f187f5e-336f-4646-bbb1-42f456c17800.png)
![image](https://user-images.githubusercontent.com/16158238/184895307-e0bed2a4-8f42-4f80-92a5-9c745ccd763a.png)
![image](https://user-images.githubusercontent.com/16158238/184895280-0e7639e0-8b50-4afd-a1a3-7d804886e6f2.png)

Expect:
https://api.maptiler.com/maps/basic-v2/?key=bkB2QxUTpyeGpXpySRpi#17.79/16.442555/107.591512/-26.6
<img width="465" alt="Screen Shot 2022-08-16 at 20 40 19" src="https://user-images.githubusercontent.com/16158238/184894015-af8cced9-dabd-44ad-955d-5cd50d599d28.png">
https://api.maptiler.com/maps/basic-v2/?key=bkB2QxUTpyeGpXpySRpi#16.86/16.464592/107.583731/-41
<img width="770" alt="image" src="https://user-images.githubusercontent.com/16158238/184896238-f1db5edf-d1b1-496d-b13b-14abead9211b.png">
https://api.maptiler.com/maps/basic-v2/?key=bkB2QxUTpyeGpXpySRpi#16.76/16.479278/107.576668/-26.6
<img width="600" alt="image" src="https://user-images.githubusercontent.com/16158238/184894618-14351706-4b85-476b-a1ac-9e54f8c6fec7.png">

Here is my code:

```
UrlTileSource tileSource = OpenMapTilesMvtTileSource.builder()
        .apiKey("bkB2QxUTpyeGpXpySRpi") // Put
        .httpFactory(new OkHttpEngine.OkHttpFactory())
        .locale("vi")
        .build();
VectorTileLayer l = mMap.setBaseMap(tileSource);
mMap.setTheme(VtmThemes.OPENMAPTILES);
mMap.layers().add(new BuildingLayer(mMap, l));
mMap.layers().add(new LabelLayer(mMap, l));
TileGridLayer mGridLayer = new TileGridLayer(mMap);
mMap.layers().add(mGridLayer);
mMap.setMapPosition(new MapPosition(16.4479445, 107.5957691, 1 << 16));
```

Thank you for your work!

**Comment by devemux86** [comment]
Probably the [OpenMapTiles](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/openmaptiles.xml) theme in VTM needs improvements.

Most users focus on offline maps, so online maps have little support.

The current OpenMapTiles theme is for demonstration purposes, needs work.

**Reply by PhungVanHoa** [reply]
Can you give me some suggestions. i will create a pull request when it fixed

**Comment by devemux86** [comment]
You need to study the [OpenMapTiles schema](https://openmaptiles.org/schema/) (as they change the OpenStreetMap tags),
the [render theme documentation](https://github.com/mapsforge/vtm/blob/master/docs/Rendertheme.md) and the [OpenMapTiles theme](https://github.com/mapsforge/vtm/blob/master/vtm-themes/resources/assets/vtm/openmaptiles.xml) and test the results.

**Reply by PhungVanHoa** [reply]
Thank you. 
Last question. Editing themes can make the dashed line close properly ?

**Comment by devemux86** [comment]
> Editing themes can make the dashed line close properly ?

We cannot know from an image (could be OpenGL artifact).

You need to check the map element and the theme rules at that area,
see if another map element or theme rule overlaps / hides it, etc.

Writing render themes requires a lot of testing.
